lock-api/src/main/java/com/palantir/lock/v2/NamespacedTimelockRpcClient.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.v2;
+
+import java.util.Set;
+
+import com.palantir.lock.client.IdentifiedLockRequest;
+import com.palantir.timestamp.TimestampRange;
+
+/**
+ * Adapter that mirrors {@code TimelockRpcClient}, but for convenience automatically provides the namespace as a
+ * parameter.
+ */
+
```

Comment:
nit: extra space.

Reply:
I'm very conflicted, the naming scheme in part 8 seems/feels more intuitive if inverted here. It feels natural to be like "I'm going to pass around a `TimelockRpcClient`" and the actual thing that has the namespace is `NamespaceAwareTimelockRpcClient`. Why I think the other one works is because we never actually use the intermediate types/adapters bar the top level `RemoteLockServiceAdaptor` one which implements `LockService`. But here, we specifically *do* want to use the middle layer i.e. this class. Then when using it, it's like "what's the namespace? I haven't had to pass one in when using any of the methods on it."

So in a sense, it makes a bit of sense for the base layer to be `NamespaceAwareTimelockRpcClient` and for this class to be `TimelockRpcClient`. Happy for either to go in, just feeling somewhat uneasy.

Reply:
discussed offline: this is named `Namespaced` because the namespace is already provided (references to `NamespaceAware` were accidental left-overs, which I will clean up) - as discussed will go through with that. I think it would be nice to keep the thing that actually makes the RPC calls to be `TimelockRpcClient`.